src/tar.ts

```diff
@@ -58,10 +79,10 @@ export async function createTar(
     const args = [
         "-cz",
         "-f",
-        CacheFilename,
+        CacheFilename?.replace(/\\/g, "/"),
```

Comment:
Why are these file names optional?
If they're `undefined` we should throw a warning, but at this point in the code, I would expect both of them to be defined since we've already validated our inputs.

Reply:
Removed. 

I kept those in tests because otherwise the testes would fail with `workspace is possibly undefined` since we have strict null check enabled.